bubblewrap.c

```diff
@@ -2463,8 +2480,8 @@ main (int    argc,
       prctl (PR_SET_SECCOMP, SECCOMP_MODE_FILTER, &seccomp_prog) != 0)
     die_with_error ("prctl(PR_SET_SECCOMP)");
 
-  if (execvp (argv[0], argv) == -1)
-    die_with_error ("execvp %s", argv[0]);
+  if (execvp (opt_argv0, argv) == -1)
```

Comment:
This is backwards...the point of providing `argv0` is to the *child* process so it can discover its personality.  Look at what e.g. the `-a` option for bash's `exec -a foo progname ...` does.

I thought `/usr/bin/egrep` was the classic example here but it looks like at least in the Fedora userspace it's just:
```
#!/bin/sh
exec grep -E "$@"
```

Bash itself behaves differently if invoked as `/bin/sh`, but I dunno offhand how to demo the difference between posix and bash mode.

And I completely forget now honestly why I filed a bug for this issue.

But anyways I'm pretty sure this approach is wrong.

Reply:
(Or well, you could say that the current logic is right, and the option should be named `--path`, and then the first non-option argument becomes `argv0`, but if that's the case the naming of things are wrong, and I'm still not sure that the arg processing is right)

Reply:
OK, I just read back into the original issue and I had the naming of things wrong there too.  If one looks at `man execve`, we'd call this option `--filename` probably, or perhaps to be ultra clear, `--exec-filename`?

Reply:
@cgwalters ok, sounds good for me. So, after implementing --exec-filename, should we still support the "last argument as argv0" that we currently have?

Reply:
There is no point in separating "what I am going to exec" and "what I'm going to put in argv[0]" unless you are going to set them to distinct values.

The use-case is to do things like "execute bash, but make it think it's sh".

```
# Common case:
# argv[] = { "/usr/bin/python3", "-c", "...", NULL }
# execve(argv[0], argv, environ)
perl -e 'system("/usr/bin/python3", "-c", "import sys; print(sys.executable)")'                                
/usr/bin/python3
# Rare case:
# argv[] = { "/pretending/not/to/be/python", "-c", "...", NULL }
# execve("/usr/bin/python3", argv, environ)
perl -e 'system {"/usr/bin/python3"} ("/pretending/not/to/be/python", "-c", "import sys; print(sys.executable)")'
/pretending/not/to/be/python
```

So this makes no sense unless you use it like this:

```
bwrap ... --unshare-pid --exec-filename ls i-cant-believe-its-not-ls -la
```

which would end up as

```
argv[] = {"i-cant-believe-its-not-ls", "-la", NULL}
execve("ls", argv, environ)
```

Reply:
@smcv that sounds good. Will send a new PR with the suggested changes.

Reply:
@smcv @cgwalters hope to address all your comments. In the same commit, I changed the completion file, aiming the new parameters. Let me know if you have concerns about the change.